src/midi_file_processor.py

```diff
@@ -0,0 +1,149 @@
+
+        char_to_midi = self._get_note_mapping()
+
+        i = 0
+        while i < len(user_input):
+            char = user_input[i]
+            if user_input.startswith(Commands.INC_BPM_80_UNITS.value.lower(), i):
+                self.bpm = self._adjust_bpm(user_input, i, self.bpm)
+                microseconds_per_beat = mido.bpm2tempo(self.bpm)
+                self.track.append(mido.MetaMessage('set_tempo', tempo=microseconds_per_beat))
+                i += len(Commands.INC_BPM_80_UNITS.value)
+            elif user_input.startswith(Commands.DEC_BPM_80_UNITS.value.lower(), i):
+                self.bpm = self._adjust_bpm(user_input, i, self.bpm, increase=False)
+                microseconds_per_beat = mido.bpm2tempo(self.bpm)
+                self.track.append(mido.MetaMessage('set_tempo', tempo=microseconds_per_beat))
+                i += len(Commands.DEC_BPM_80_UNITS.value)
+            elif char in Commands.notes():
+                self._process_note(char_to_midi, char)
+                i += 1
+            elif char == Commands.SILENCE.value:
+                self._process_pause()
+                i += 1
+            elif char == Commands.DOUBLE_VOLUME.value:
+                self.volume = min(127, self.volume + self.volume)
+                i += 1
+            elif char == Commands.RESET_VOLUME.value:
+                self.volume = self._DEFAULT_VOLUME
+                i += 1
+            elif char in Commands.repeat_commands():
+                self._process_instrument_change(user_input, i, char_to_midi)
+                i += 1
+            elif user_input.startswith(Commands.INC_1_OCTAVE.value, i):
+                self.octave = min(self._MAX_OCTAVE, self.octave + 12)
+                i += len(Commands.INC_1_OCTAVE.value)
+            elif user_input.startswith(Commands.DEC_1_OCTAVE.value, i):
+                self.octave = max(self._MIN_OCTAVE, self.octave - 12)
+                i += len(Commands.DEC_1_OCTAVE.value)
+            elif char == Commands.RANDOM_NOTE.value:
+                self._process_random_note()
+                i += 1
+            elif char == Commands.CHANGE_INSTRUMENT.value:
+                self._process_new_line()
+                i += 1
+            elif char == Commands.RANDOM_BPM.value:
+                self.bpm = random.randint(40, 200)
+                microseconds_per_beat = mido.bpm2tempo(self.bpm)
+                self.track.append(mido.MetaMessage('set_tempo', tempo=microseconds_per_beat))
+                i += 1
+            else:
+                i += 1  # Ignora caracteres inválidos
+
+        self._save_midi_file(parent_widget)
+
+    def _get_note_mapping(self):
+        return {
+            Commands.NOTE_LA.value: 69,
+            Commands.NOTE_SI.value: 71,
+            Commands.NOTE_DO.value: 60,
+            Commands.NOTE_RE.value: 62,
+            Commands.NOTE_MI.value: 64,
+            Commands.NOTE_FA.value: 65,
+            Commands.NOTE_SOL.value: 67,
+        }
```

Comment:
Não precisa do `.value`.

```suggestion
            Commands.NOTE_LA: 69,
            Commands.NOTE_SI: 71,
            Commands.NOTE_DO: 60,
            Commands.NOTE_RE: 62,
            Commands.NOTE_MI: 64,
            Commands.NOTE_FA: 65,
            Commands.NOTE_SOL: 67,
        }
```